models/CMakeLists.txt

```diff
@@ -1,19 +1,30 @@
-# List the subdirectory (http://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory)
-macro(SUBDIRLIST result curdir)
-  file(GLOB children RELATIVE ${curdir} ${curdir}/*)
-  set(dirlist "")
-  foreach(child ${children})
-    if(IS_DIRECTORY ${curdir}/${child})
-      list(APPEND dirlist ${child})
-    endif()
-  endforeach()
-  set(${result} ${dirlist})
-endmacro()
-
-# Get list of models 
-subdirlist(subdirs ${CMAKE_CURRENT_SOURCE_DIR})
-
-# Install each model 
-foreach (dir ${subdirs})
-  install (DIRECTORY ${dir} DESTINATION share/gazebo/models)
-endforeach ()
\ No newline at end of file
+# configure_files for all files in a folder
+# https://stackoverflow.com/questions/697560/how-to-copy-directory-from-source-tree-to-binary-tree
+macro(CONFIGURE_FILES srcDir destDir)
+  make_directory(${destDir})
+  file(GLOB_RECURSE templateFiles RELATIVE ${srcDir} ${srcDir}/*)
```

Comment:
It depends on much in depth you want to sweep the folder, `GLOB` enters only in the first level childs, `GLOB_RECURSE` sweeps the entire tree. Here I need to reach the leafs.